pkg/scaling/executor/scale_scaledobjects.go

```diff
@@ -28,6 +29,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	kedacontrollerutil "github.com/kedacore/keda/v2/controllers/keda/util"
```

Comment:
Just my thinking, we should unify the functions for Status & Conditions handling. Referencing controller related stuff here is suboptimal.

There are two locations with similar functions: 
https://github.com/kedacore/keda/blob/9f0ee58fc6839d74636e1ec6883fe21d82617ea3/controllers/keda/util/status.go#L30-L63
https://github.com/kedacore/keda/blob/9f0ee58fc6839d74636e1ec6883fe21d82617ea3/pkg/scaling/executor/scale_executor.go#L64-L132

We should refactor this to have it in one place.

Reply:
We could move `keda/controllers/keda/util/status.go` to `pkg/util/status.go`, this way the `SetStatusConditions` and `UpdateScaldObjectStatus` can be used across the project. As for unifying the functions for Status and Conditions handling, I think that should be tackled in a different PR?

Reply:
Yeah a change like that. And I concur this should be tackled in a different PR.